src/wp-includes/class-wp-image-editor.php

```diff
@@ -336,11 +337,17 @@ protected function get_output_format( $filename = null, $mime_type = null ) {
 			// If no file specified, grab editor's current extension and mime-type.
 			$file_ext  = strtolower( pathinfo( $this->file, PATHINFO_EXTENSION ) );
 			$file_mime = $this->mime_type;
+			// Favor the file mime extension.
+			$ext = $this->get_extension( $file_mime );
+			if ( $this->mime_type_set && $ext !== $file_ext ) {
+				$new_ext   = $ext;
+				$mime_type = $file_mime;
+			}
```

Comment:
Instead of handling that here, I think we need to check if `$this->mime_type_set` right at the top of this method. Basically, that should only be relevant if the `$mime_type` method parameter is `null`, it should basically become the default value then.

Somewhat like this should be on top of the method:
```php
if ( ! $mime_type && $this->mime_type_set ) {
	$mime_type = $this->mime_type;
}
```

I think that's all we need to do to make this work as expected.

Essentially, the output MIME type should be determined as follows in this method:
1. If `$mime_type` parameter is provided, use that.
2. Else if `$this->mime_type_set`, use the set MIME type, as it was manually set. This is the new piece, essentially it is a fallback for `$mime_type`, used when the method is called without a `$mime_type`, but still one was _manually_ set.
3. Else use the `$file_mime`, similar to today.